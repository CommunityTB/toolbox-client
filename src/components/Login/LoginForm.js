import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
import { Button, Input } from '../../helpers/Helpers';

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = event => {
    event.preventDefault();
    const { user_name, user_password } = event.target;

    AuthApiService.postLogin({
      user_name: user_name.value,
      user_password: user_password.value,
    })
    .then(response => {
      user_name.value=''
      user_password.value=''
      TokenService.saveAuthToken(response.authToken)
      this.props.onLoginSuccess(response.userId)
    })
    .catch(response => {
      this.setState({ error: response.error})
    })
  }

  render() {
    const { error } = this.state;
    return(
      <section>
        <div className='form-space'>
          <form 
            className='login-form'
            onSubmit={this.handleSubmitJwtAuth}
          >
            <legend className='page-title'>Login</legend> 
            <label htmlFor='user_name' className='user-name'>Username</label>
            <Input
                type='text' 
                id='user_name'
                name='userName' 
                required>
            </Input> 
            
            <label htmlFor='user_password' className='user-password'>Password</label>       
            <Input   
              type='password' 
              id='user_password'
              name='user_password' 
              required>
            </Input>   
            
            <Button type='submit' className='login-button'>
                Login
            </Button>

            <Link className="cancel-action" to='/'>Cancel</Link>
            <Link className="cancel-action" to='/register'>Need an account?</Link>
            <div role='alert'>
              {error && <p className='red-error'>{error}</p>}
            </div>
          </form>
        </div>
    </section>
    )
  }
}

