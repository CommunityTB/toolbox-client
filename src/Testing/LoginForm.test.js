import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from '../components/Login/LoginForm'

describe(`LoginForm component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><LoginForm /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a LoginForm by default', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})