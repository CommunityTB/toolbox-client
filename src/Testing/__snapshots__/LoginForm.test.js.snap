// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a LoginForm by default 1`] = `
<section>
  <div
    className="form-space"
  >
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <legend
        className="page-title"
      >
        Login
      </legend>
      <label
        className="user-name"
        htmlFor="user_name"
      >
        Username
      </label>
      <Input
        className="form-input-field"
        id="user_name"
        name="userName"
        required={true}
        type="text"
      />
      <label
        className="user-password"
        htmlFor="user_password"
      >
        Password
      </label>
      <Input
        className="form-input-field"
        id="user_password"
        name="user_password"
        required={true}
        type="password"
      />
      <Button
        className="login-button"
        type="submit"
      >
        Login
      </Button>
      <Link
        className="cancel-action"
        to="/"
      >
        Cancel
      </Link>
      <Link
        className="cancel-action"
        to="/register"
      >
        Need an account?
      </Link>
      <div
        role="alert"
      />
    </form>
  </div>
</section>
`;
